datarobot_batch_scoring/batch_scoring.py

```diff
@@ -134,7 +137,7 @@ def run_batch_predictions(base_url, base_headers, user, pwd,
                     [c for c in keep_cols if c not in first_row.fieldnames],
                     first_row.fieldnames))
 
-        if not dry_run:
+        if not (dry_run or import_id):
```

Comment:
Probably it's better to move import_id in next check. To not mix dry_run processing and missing data fix...